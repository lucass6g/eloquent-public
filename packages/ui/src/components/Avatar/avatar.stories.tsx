import { Meta, StoryObj } from "@storybook/react";
import { Avatar } from ".";

const meta = {
  title: "Avatar",
  render: () => (
    <Avatar.Root>
      <Avatar.Image src="https://github.com/shadcn.png" alt="@shadcn" />
      <Avatar.Fallback>Shadcn</Avatar.Fallback>
    </Avatar.Root>
  ),
  tags: ["autodocs"],
  parameters: {
    layout: "centered",
  },
} satisfies Meta<{}>;

export default meta;

type Story = StoryObj<typeof meta>;

export const AvatarWithImage: Story = {
  args: {},
  render: () => (
    <Avatar.Root>
      <Avatar.Image src="https://github.com/shadcn.png" alt="@shadcn" />
      <Avatar.Fallback>Avatar Shad</Avatar.Fallback>
    </Avatar.Root>
  ),
};
export const AvatarWithFallback: Story = {
  args: {},
  render: () => (
    <Avatar.Root>
      <Avatar.Image src="" alt="@shadcn"/>
      <Avatar.Fallback>Elo</Avatar.Fallback>
    </Avatar.Root>
  ),
};
export const AvatarWithSizesProperties: Story = {
  args: {},
  name: 'Avatar com tamanhos especÃ­ficos',
  render: () => (
    <Avatar.Root size="xl">
      <Avatar.Image src="" alt="@shadcn"/>
      <Avatar.Fallback fontSize="lg">Elo</Avatar.Fallback>
    </Avatar.Root>
  ),
};
