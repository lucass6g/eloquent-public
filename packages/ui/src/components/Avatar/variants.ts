import {tv} from "tailwind-variants";

export const avatarRootVariants = tv({
    base: [
        "relative flex shrink-0 overflow-hidden rounded-full"
    ],
    variants: {
        size: {
            xsm: "h-8 w-8",
            sm: 'h-10 w-10',
            md:"h-16 w-16",
            lg: "h-20 w-20",
            xl: 'h-24 w-24',
            huge: 'h-[104px] w-[104px]'
        },
    },
    defaultVariants: {
        size: "md",
    },
})

export const avatarFallbackVariants = tv({
    base: [
        "flex h-full w-full items-center justify-center rounded-full bg-muted"
    ],
    variants: {
        fontSize: {
            sm: 'text-sm',
            md: 'text-base',
            lg: 'text-lg'
        }
    },
    defaultVariants: {
        fontSize: 'md'
    },
})