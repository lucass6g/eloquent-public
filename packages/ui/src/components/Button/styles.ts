import { cva } from 'class-variance-authority';

export type ClassVariantsProps = Readonly<{
	variant: {
		primary?: string[]
		secondary?: string[]
		link?: string[]
		icon?: string[]
	}
	size: {
		small?: string[]
		medium?: string[]
		large?: string[]
		icon?: string[]
	}
	inverse: {
		true?: string[]
		false?: string[]
	}
	fullWidth: {
		true?: string[]
		false?: string[]
	}
}>

export const buttonVariants = cva<ClassVariantsProps>(
	[
		'font-sans text-xs font-semibold',
		'outline-offset-4',
		'box-border border-2',
		'transition-all',
		'disabled:opacity-50 disabled:border-none disabled:cursor-not-allowed ',
	],
	{
		variants: {
			variant: {
				primary: [
					'text-neutral-100',
					'border-transparent',
					'bg-positive-100',
					'hover:bg-neutral-200',
					'active:bg-base-light-01 active:text-base-dark-01',
				],
				secondary: [
					'bg-neutral-dark-100',
					'text-neutral-dark-1000',
					'border-base-light-02',
					'hover:bg-base-light-02',
					'active:border-base-light-02 bg-base-light-01',
				],
				link: ['text-primary underline-offset-4 hover:underline'],
				icon: [
					'bg-transparent border-transparent font-normal border-none',
					'flex  flex-row  gap-stack-xxs',
				],
			},
			size: {
				small: ['px-inline-xxxs h-8 leading-tiny text-xxs'],
				medium: ['px-inline-xxs h-12 leading-default'],
				large: ['px-inline-xs h-14 leading-distant text-sm'],
				icon: [],
			},
			inverse: {
				true: [''],
			},
			fullWidth: {
				true: ['w-full'],
			},
		},
		defaultVariants: {
			variant: 'primary',
			size: 'medium',
		},
		compoundVariants: [
			{
				variant: 'primary',
				inverse: true,
				class: [
					'bg-neutral-100',
					'text-neutral-1000',
					'hover:bg-base-light-02',
					'active:bg-base-light-03 active:text-base-dark-03',
				],
			},
			{
				variant: 'secondary',
				inverse: true,
				class: [
					'bg-transparent',
					'border-neutral-100',
					'text-neutral-100',
					'hover:bg-base-dark-03',
					'active:bg-base-dark-01 active:text-base-light-01',
					'disabled:bg-neutral-1000 disabled:opacity-50 disabled:border-base-light-02',
				],
			},
		],
	},
)
