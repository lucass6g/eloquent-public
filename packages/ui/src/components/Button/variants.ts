import {tv} from "tailwind-variants"

export const baseButton = tv({
    base: [
        "inline-flex items-center justify-center whitespace-nowrap",
        "border-lg rounded-lg transition-colors",
        "focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1",
        "disabled:pointer-events-none",
        "text-sm font-medium antialiased",
        "leading-4"
    ]
})

export const buttonVariants = tv({
    extend: baseButton,
    variants: {
        variant: {
            default:[
                "bg-primary text-primary-foreground shadow",
                "hover:bg-green-dark-400",
                "disabled:bg-neutral-300 disabled:text-neutral-500 disabled:font-medium",
            ],
            secondary: [
                "shadow-sm",
                "border-foreground",
                "text-accent-foreground hover:bg-accent hover:text-accent-foreground",
            ],
            destructive:
                "bg-destructive text-destructive-foreground shadow-sm hover:opacity-90",
            link: "text-primary underline-offset-4 hover:underline",
            icon: "p-0",
            loading: "",
        },
        size: {
            default: "py-2 px-4 min-w-[152px] min-h-[24px] border-lg",
            sm: "h-8 px-3 text-xs",
            icon: "h-9 w-9 first-child:stroke-primary-foreground",
        },
    },
    defaultVariants: {
        variant: "default",
        size: "default",
    },
});