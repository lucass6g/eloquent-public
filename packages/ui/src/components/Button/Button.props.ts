import { VariantProps } from "tailwind-variants";
import { buttonVariants } from "./variants";
import { ValueOf } from "src/types/select-key";

export interface ButtonHTMLAttributes extends React.ButtonHTMLAttributes<HTMLButtonElement>{}
export interface ButtonVariantsProps extends VariantProps<typeof buttonVariants>{}
export interface ButtonEloquentProps  {
    asChild?: boolean
}

export interface  ButtonProps extends ButtonEloquentProps, ButtonHTMLAttributes, ButtonVariantsProps {}

export type ButtonVariant = Required<ValueOf<ButtonVariantsProps, "variant">>;
export type ButtonSize = Required<ValueOf<ButtonVariantsProps, "size">>