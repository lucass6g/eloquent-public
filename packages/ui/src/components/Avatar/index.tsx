"use client"

import * as React from "react"
import * as AvatarPrimitive from "@radix-ui/react-avatar"


import { type VariantProps } from "class-variance-authority";

import { cn } from "@eloquent/styles";
import { avatarFallbackVariants, avatarRootVariants } from "./variants";


const RootAvatar = React.forwardRef<
  React.ElementRef<typeof AvatarPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root> & VariantProps<typeof avatarRootVariants>
>(({ className, size, ...props }, ref) => (
  <AvatarPrimitive.Root
    ref={ref}
    className={cn(avatarRootVariants({ size }), className)}
    {...props}
  />
))
RootAvatar.displayName = AvatarPrimitive.Root.displayName

const AvatarImage = React.forwardRef<
  React.ElementRef<typeof AvatarPrimitive.Image>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>
>(({ className, ...props }, ref) => (
  <AvatarPrimitive.Image
    ref={ref}
    className={cn("aspect-square h-full w-full", className)}
    {...props}
  />
))
AvatarImage.displayName = AvatarPrimitive.Image.displayName

const AvatarFallback = React.forwardRef<
  React.ElementRef<typeof AvatarPrimitive.Fallback>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback> & VariantProps<typeof avatarFallbackVariants>
>(({ className,fontSize, ...props }, ref) => (
  <AvatarPrimitive.Fallback
    ref={ref}
    className={cn(avatarFallbackVariants({ fontSize }),
      className
    )}
    {...props}
  />
))
AvatarFallback.displayName = AvatarPrimitive.Fallback.displayName

export const Avatar = {
    Root: RootAvatar,
    Image:AvatarImage,
    Fallback:AvatarFallback
}
