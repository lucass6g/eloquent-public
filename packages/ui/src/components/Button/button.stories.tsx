import { Button } from ".";
import type { Meta, StoryObj } from "@storybook/react";
import { LayoutGrid, X } from "lucide-react";

import { ButtonProps, ButtonSize, ButtonVariant } from "./Button.props";

const meta: Meta<ButtonProps> = {
  title: "Button",
  component: Button,
  args: {
    children: "Clique aqui",
    disabled: false,
    size: "default",
    loading: false
  },
  argTypes: {
    onClick: {
      action: "onClick",
      table: {
        disable: true,
      },
    },
    onFocus: {
      action: "onFocus",
      table: {
        disable: true,
      },
    },
    onMouseOver: {
      action: "onMouseOver",
      table: {
        disable: true,
      },
    },
    type: {
      control: { type: "inline-radio" },
      options: ["button", "submit", "reset"],
      description: "Tipo do botão",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "button" },
      },
    },
    variant: {
      control: { type: "select" },
      options: [
        "default",
        "destructive",
        "link",
        "secondary",
        "badge",
        "none"
      ] satisfies ButtonVariant[],
      description: "Variante do botão",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "default" },
      },
    },
    size: {
      control: { type: "select" },
      options: ["default", "icon", "sm", "none"] satisfies ButtonSize[],
      description: "Tamanho do botão",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "default" },
      },
    },
    disabled: {
      description: "Se o botão está desabilitado",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    loading: {
      description: "Se o botão está em estado de `carregando...`",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    labelLoading: {
      description: "Texto que aparece quando o botão está em estado de carregando",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "Carregando..." },
      },
    }
  },
  tags: ["experimental"]
};

export default meta;

type Story = StoryObj<ButtonProps>;

export const Primary: Story = {
  args: {
    variant: "default",
  },
};

export const Secondary: Story = {
  args: {
    variant: "secondary",
  },
};

export const Link: Story = {
  args: {
    variant: "link",
  },
};

export const Destructive: Story = {
  args: {
    variant: "destructive",
  },
};

export const NoStyle: Story = {
  args: {
    variant: "none",
    size: "none",
  },
};

export const WithIcon: Story = {
  render(props) {
    return (
        <Button {...props}>
          <LayoutGrid className="w-4 h-4 mr-2"/>
          Clique aqui
        </Button>
    );
  }
};

export const Loading: Story = {
  render: (props) => {
    return (
      <Button {...props}>
        Clique aqui
      </Button>
    );
  },
  args: {
    loading: true
  }
}

export const Badge: Story = {
  render: (props) => {
    return (
          <Button {...props} variant="badge">
            Clique aqui
            <X className="ml-2 w-4 h-4" />
          </Button>
    );
  },
  args: {
    variant: "badge"
  }
}

export const Icon: Story = {
  render: ({children, ...props}) => {
    return(
      <Button {...props} size="icon" variant="icon">
        <LayoutGrid className="h-4 w-4" />
      </Button>
    )
  }
};
