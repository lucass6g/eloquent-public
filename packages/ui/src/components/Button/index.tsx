import * as React from "react";
import {Slot} from "@radix-ui/react-slot";
import {cn} from "@eloquent/styles";
import { ButtonProps } from "./Button.props";
import { buttonVariants } from "./variants";
import { Loader2 } from "lucide-react";

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
    function EloquentButton({
        className, 
        variant = "default", 
        disabled = false, 
        loading = false, 
        size = "default", 
        asChild = false, 
        children,
        labelLoading = "Carregando...",
        ...props
    }, 
        ref)  {
        const Comp = asChild ? Slot : "button";

        const LoadingContent = (
            <div className="flex items-center">
                <Loader2 className={cn("animate-spin h-4 w-4 mr-2", size === "icon" && "m-auto")}/>
                {size !== "icon" && labelLoading}
            </div>
        )
          
        return (
            <Comp
                disabled={disabled || loading}
                className={cn(variant === "none" ? buttonVariants({variant, className}): 
                buttonVariants({variant, size, className}))}
                ref={ref}
                {...props}
        >
            {loading ? LoadingContent : children}
        </Comp>
    );
    }
);
