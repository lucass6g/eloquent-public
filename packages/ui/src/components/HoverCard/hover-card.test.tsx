import { render, screen, waitFor } from '@testing-library/react';
import { it, describe, expect } from "vitest"
import userEvent from '@testing-library/user-event';
import { HoverCard } from '.';
import { Button } from '../Button';
import { Avatar } from '../Avatar';
import { CalendarDays } from 'lucide-react';


describe('HoverCard Component', () => {
    it('renders HoverDefault story and displays content on hover', async () => {
        const component = render(<HoverCard.Root>
            <HoverCard.Trigger>
                <Button variant="link">@Cartão Elo</Button>
            </HoverCard.Trigger>
            <HoverCard.Content>
                Cartão Elo - O cartão do Brasileiro.
            </HoverCard.Content>
        </HoverCard.Root>);

        const triggerButton = screen.getByText('@Cartão Elo');
        expect(triggerButton).toBeInTheDocument();

        await userEvent.hover(triggerButton);
        const content = await screen.findByText('Cartão Elo - O cartão do Brasileiro.');
        expect(content).toBeInTheDocument();
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders HoverCardExample story and displays content with avatar on hover', async () => {
        const component = render(
            <HoverCard.Root>
                <HoverCard.Trigger asChild>
                    <Button variant="link">@Cartão Elo</Button>
                </HoverCard.Trigger>
                <HoverCard.Content className="w-80">
                    <div className="flex space-x-4">
                        <Avatar.Root size='sm'>
                            <Avatar.Fallback data-testid="avatar">Elo</Avatar.Fallback>
                        </Avatar.Root>
                        <div className="space-y-1">
                            <h4 className="text-sm font-semibold">Cartão Elo</h4>
                            <p className="text-sm">O cartão do Brasileiro.</p>
                            <div className="flex items-center pt-2">
                                <CalendarDays className="mr-2 h-4 w-4 opacity-70" />
                                <span className="text-xs text-muted-foreground">
                                    Fundada em 2011
                                </span>
                            </div>
                        </div>
                    </div>
                </HoverCard.Content>
            </HoverCard.Root>
        );
        const triggerButton = screen.getByText('@Cartão Elo');
        expect(triggerButton).toBeInTheDocument();
        userEvent.hover(triggerButton);
        await waitFor(() => {
            const avatarFallback = screen.getByTestId('avatar');
            expect(avatarFallback).toBeInTheDocument();
            const content = screen.getByText('Cartão Elo');
            expect(content).toBeInTheDocument();
            const subContent = screen.getByText('O cartão do Brasileiro.');
            expect(subContent).toBeInTheDocument();
            const calendarText = screen.getByText('Fundada em 2011');
            expect(calendarText).toBeInTheDocument();
            expect(component.baseElement).toMatchSnapshot();
        });
    });
});