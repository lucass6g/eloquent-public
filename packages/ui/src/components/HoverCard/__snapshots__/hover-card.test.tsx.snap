// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HoverCard Component > renders HoverCardExample story and displays content with avatar on hover 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-sm font-medium antialiased leading-4 text-primary underline-offset-4 hover:underline py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
      data-state="open"
    >
      @Cartão Elo
    </button>
    <div
      data-radix-popper-content-wrapper=""
      style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content;"
    >
      <div
        class="z-50 rounded-[8px] border bg-popover p-4 text-popover-foreground shadow-md outline-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 w-80"
        data-align="center"
        data-side="bottom"
        data-state="open"
        style="--radix-hover-card-content-transform-origin: var(--radix-popper-transform-origin); --radix-hover-card-content-available-width: var(--radix-popper-available-width); --radix-hover-card-content-available-height: var(--radix-popper-available-height); --radix-hover-card-trigger-width: var(--radix-popper-anchor-width); --radix-hover-card-trigger-height: var(--radix-popper-anchor-height); animation: none;"
      >
        <div
          class="flex space-x-4"
        >
          <span
            class="relative flex shrink-0 overflow-hidden rounded-full h-10 w-10"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-muted text-base"
              data-testid="avatar"
            >
              Elo
            </span>
          </span>
          <div
            class="space-y-1"
          >
            <h4
              class="text-sm font-semibold"
            >
              Cartão Elo
            </h4>
            <p
              class="text-sm"
            >
              O cartão do Brasileiro.
            </p>
            <div
              class="flex items-center pt-2"
            >
              <svg
                class="lucide lucide-calendar-days mr-2 h-4 w-4 opacity-70"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2v4"
                />
                <path
                  d="M16 2v4"
                />
                <rect
                  height="18"
                  rx="2"
                  width="18"
                  x="3"
                  y="4"
                />
                <path
                  d="M3 10h18"
                />
                <path
                  d="M8 14h.01"
                />
                <path
                  d="M12 14h.01"
                />
                <path
                  d="M16 14h.01"
                />
                <path
                  d="M8 18h.01"
                />
                <path
                  d="M12 18h.01"
                />
                <path
                  d="M16 18h.01"
                />
              </svg>
              <span
                class="text-xs text-muted-foreground"
              >
                Fundada em 2011
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`HoverCard Component > renders HoverDefault story and displays content on hover 1`] = `
<body>
  <div>
    <a
      data-state="open"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-sm font-medium antialiased leading-4 text-primary underline-offset-4 hover:underline py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
      >
        @Cartão Elo
      </button>
    </a>
    <div
      data-radix-popper-content-wrapper=""
      style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 4px); min-width: max-content; --radix-popper-available-width: 0px; --radix-popper-available-height: -4px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 50% 0px;"
    >
      <div
        class="z-50 w-64 rounded-[8px] border bg-popover p-4 text-popover-foreground shadow-md outline-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2"
        data-align="center"
        data-side="bottom"
        data-state="open"
        style="--radix-hover-card-content-transform-origin: var(--radix-popper-transform-origin); --radix-hover-card-content-available-width: var(--radix-popper-available-width); --radix-hover-card-content-available-height: var(--radix-popper-available-height); --radix-hover-card-trigger-width: var(--radix-popper-anchor-width); --radix-hover-card-trigger-height: var(--radix-popper-anchor-height);"
      >
        Cartão Elo - O cartão do Brasileiro.
      </div>
    </div>
  </div>
</body>
`;
